### PR TITLE
Add support for GHC 9.4.1

### PR DESCRIPTION
**Pull request by christiaanb**

### CODE REVIEW

File: tests/ErrorTests.hs [hunk 1]
```diff
@@ -250,7 +261,9 @@ testProxy13 :: Proxy (4*a) -> Proxy (2*a) ->()
 testProxy13 = proxyInEq
 
 testProxy13Errors =
-#if __GLASGOW_HASKELL__ >= 902
+#if __GLASGOW_HASKELL__ >= 904
+  ["Cannot satisfy: 4 * a <= 2 * a"]
```
Comment (martijnbastiaan) [inline tests/ErrorTests.hs:265]: Oh hey, does this mean we get better type errors from 9.4 onwards?

Comment (christiaanb) [reply tests/ErrorTests.hs:265]: Yes. We also get `Cannot satisfy: 2 <= 1` as opposed to `Couldn't match type 'False with 'True`

Comment (martijnbastiaan) [reply tests/ErrorTests.hs:265]: Very cool

File: cabal.project [hunk 2]
```diff
@@ -2,3 +2,8 @@ packages: ./ghc-typelits-natnormalise.cabal
 
 package ghc-typelits-natnormalise
     flags: +deverror
+
+source-repository-package
+  type: git
+  location: https://github.com/clash-lang/ghc-tcplugins-extra.git
+  tag: 8852d8ad94effa39e3400a243b21ae444df6c69e
```
Comment (Mikolaj) [inline cabal.project:9]: Given that the new ghc-tcplugins-extra is already released, this can be removed, right?

**Review by Mikolaj — COMMENTED** [review]
Looks great to my very untrained eye.

**Comment by Mikolaj** [comment]
Oh, I miseed the merge. Only now it refreshed. :D Hurray!

**Comment by Bodigrim** [comment]
@christiaanb could this be released please?

**Pull request merged**